src/Compilers/CSharp/Portable/Parser/LanguageParser_InterpolatedString.cs

```diff
@@ -187,10 +188,24 @@ ReadOnlySpan<char> getIndentationWhitespace(ReadOnlySpan<char> originalTextSpan)
                             var isLegalBlankLine = isBlankLine && indentationWhitespace.StartsWith(currentLineWhitespace);
                             if (!isLegalBlankLine)
                             {
-                                indentationError ??= MakeError(
-                                    lineStartPosition,
-                                    width: currentIndex - lineStartPosition,
-                                    ErrorCode.ERR_LineDoesNotStartWithSameWhitespace);
+                                // Specialized error message if this is a spacing difference.
+                                if (CheckForSpaceDifference(
+                                        currentLineWhitespace, indentationWhitespace,
+                                        out var currentLineWhitespaceChar, out var indentationWhitespaceChar))
+                                {
+                                    indentationError ??= MakeError(
+                                        lineStartPosition,
+                                        width: currentIndex - lineStartPosition,
+                                        ErrorCode.ERR_LineContainsDifferentWhitespace,
+                                        currentLineWhitespaceChar, indentationWhitespaceChar);
+                                }
+                                else
+                                {
+                                    indentationError ??= MakeError(
+                                        lineStartPosition,
+                                        width: currentIndex - lineStartPosition,
+                                        ErrorCode.ERR_LineDoesNotStartWithSameWhitespace);
```

Comment:
I didn't follow, why do we have the same logic and reporting the same error codes in the lexer and the parser?

Reply:
because they fundamentally operate on different types.  one sits on StringBuilder (the lexer abstraction), while one works on `ReadOnlySpan<char>` (wince the parser uses System.String for interpolations).

I don't love this, and there is a lot of duplication in lexing/parsing for the *entire* feature because of this.  but i'm not sure how to workaround that.  A core problem is simply that the lexer uses this TextWindow abstraction, so there is no guarantee of any contiguity...